grisha.pletin/practical2/notification_module.rb

```diff
@@ -0,0 +1,38 @@
+module Notification
+  def self.included(base)
+    base.extend(ClassMethods)
+  end
+
+  def send_message(recipient)
+    puts "Sending #{self.class.name} to #{recipient}"
+    check_recipient(recipient)
+    yield if block_given?
+  rescue StandardError
+    puts 'An error has occurred, check log'
+    add_to_log(recipient)
+  else
+    nil
+  end
+
+  def add_to_log(recipient)
+    File.open("#{self.class}.log", 'a') do |file|
+      file.write("#{recipient} got an error\n")
+    end
+  end
+
+  def check_recipient(recipient)
+    raise StandartError, 'Recipient is not correct' unless valid?(recipient)
+  end
+
+  def valid?(recipient)
+    recipient.match?(/^\+\d{12}/) || recipient.match?(/\A[^@\s]+@([^@\s]+\.)+[^@\s]+\z/)
```

Comment:
This  validation will not work in case a phone number is passed instead of email im Email.new.send_message

Reply:
fix